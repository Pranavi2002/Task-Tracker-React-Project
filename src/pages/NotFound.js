import React, {useContext} from "react";
// import { ThemeContext } from "../context/ThemeContext";
import useTheme from "../hooks/useTheme";

function NotFound() {
  // const { theme } = useContext(ThemeContext);
  const { theme } = useTheme();

  const style = {
    padding: "20px",
    minHeight: "70vh",
    backgroundColor: theme === "light" ? "#fff" : "#333",
    color: theme === "light" ? "#222" : "#f4f4f9",
    transition: "0.3s",
    textAlign: "center",
  };

  return (
    <div style={style}>
      <h1>404 - Page Not Found</h1>
      <p>Oops! The page you're looking for doesn't exist.</p>
    </div>
  );
}

export default NotFound;