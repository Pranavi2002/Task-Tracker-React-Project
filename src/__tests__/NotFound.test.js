// src/__tests__/NotFound.test.js
import React from "react";
import { render } from "@testing-library/react";
import NotFound from "../pages/NotFound";

// Mock useTheme hook
jest.mock("../hooks/useTheme", () => () => ({ theme: "light" }));

describe("NotFound Snapshot", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot(); // Ensure 404 page structure remains stable
  });
});