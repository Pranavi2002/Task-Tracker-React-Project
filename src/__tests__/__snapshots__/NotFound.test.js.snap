// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    style="padding: 20px; min-height: 70vh; background-color: rgb(255, 255, 255); color: rgb(34, 34, 34); transition: 0.3s; text-align: center;"
  >
    <h1>
      404 - Page Not Found
    </h1>
    <p>
      Oops! The page you're looking for doesn't exist.
    </p>
  </div>
</DocumentFragment>
`;
